inverted_index.py

```diff
@@ -0,0 +1,354 @@
+            q.put(queue_item)
+
+
+        # For the last term - posting lists
+        content = term_to_write + " " + " ".join(doc_ids_to_write)
+        self.WriteToFile(self.out_postings, content + "\n", True)
+        self.WriteToFile(self.out_dict, term_to_write + "\n", True)
+
+
+    """
+        Method to Write to out_file.
+        Params: 
+            out_file: file path
+            result: Text to store in out_file
+    """
+    def WriteToFile(self, out_file, result, append = False):
+
+        if not append:
+            fw = open(out_file, 'w+')
+            fw.write(''.join(result))
+        else:
+            # Append to the end of the file
+            fw = open(out_file, 'a')
+            fw.write(''.join(result))
+
+
+
+
+    """
+        ////////////////////////////////////////
+        Methods for Search.py
+        ////////////////////////////////////////
+    """
+
+
+    """
+        Method to load the contents of dictionary.txt into self.dictionary set
+        
+        Called by constructor when search.py initialises inverted index class
+    """
+    def LoadDictionaryFromMem(self):
+        if len(self.dictionary) != 0:
+            return
+        for term in self.ReadFromFile(self.out_dict):
+            self.dictionary.add(term.rstrip('\n'))
+
+    """
+        Method to obtain the list of posting list for input term.
+
+        Params: 
+            term: term value
+    
+        Returns:
+            A list of posting list for given term
+    """
+    def GetPostingListForTerm(self, term):
+        print("Loading Posting List for term in memory...")
+
+        offset = self.GetOffset(term)
+        if offset == -1:
+            # Term not found
+            return []
+
+        line = self.ReadFromFile(self.out_postings, offset)
+
+        # remove first item in the line which contains term value and last value \n
+        return line.split(" ")[1:-1]
+
+    """
+        Method to get offset of the term in postings.txt
+        
+        The terms in posting.txt and dictionary.txt have the same ordering.
+        term_index: We deduce the line number that the term appears in the postings.txt
+        from it's position in the sorted list of dictionary set.
+        line_offset[term_index] tells us the offset value of term
+        
+        Params: 
+            term: term value
+            result: Text to store in out_file
+        Returns:
+            offset value of term in postings.txt
+    """
+    def GetOffset(self, term):
+
+        # If self.line_offset has been constructed before, we skip this process
+        if len(self.line_offset) == 0:
+            offset = 0
+            for line in self.ReadFromFile(self.out_postings):
+                self.line_offset.append(offset)
+                offset += len(line) + 1 # 1 value is for accomodating \n
+
+        try:
+            term_index = list(sorted(self.dictionary)).index(term)
```

Comment:
can we have a sorted dictionary cached so we can avoid sorting the dictionary every time